src/compiler/checker.ts

```diff
@@ -16725,7 +16725,13 @@ namespace ts {
                 // only the class declaration node will have the Abstract flag set.
                 const valueDecl = expressionType.symbol && getClassLikeDeclarationOfSymbol(expressionType.symbol);
                 if (valueDecl && hasModifier(valueDecl, ModifierFlags.Abstract)) {
-                    error(node, Diagnostics.Cannot_create_an_instance_of_the_abstract_class_0, declarationNameToString(getNameOfDeclaration(valueDecl)));
+                    const name = getNameOfDeclaration(valueDecl);
+                    if (name) {
+                        error(node, Diagnostics.Cannot_create_an_instance_of_the_abstract_class_0, declarationNameToString(name));
+                    }
+                    else {
+                        error(node, Diagnostics.Cannot_create_an_instance_of_an_abstract_class);

```

Comment:
Alternatively, we can just use this diagnostic and remove the named one.

Reply:
That's probably better since it means having only one diagnostic and it's obvious from the error location what you're trying to construct. Also, it might not be an identifier so we can't reliably get a name from the use site either.